harness/determined/_checkpointing.py

```diff
@@ -0,0 +1,62 @@
+import logging
+from typing import Any, Dict, Optional
+
+import determined as det
+from determined.experimental import client
+
+log = logging.getLogger("generic")
```

Comment:
Yeah, `determined.generic` is _definitely_ the right namespace.

The overall plan is that this logger defaults to not printing anything (although Liam requested that we log the reported training metrics and validation metrics), but that users can enable it with normal python logging operations for debugging their use of the Generic API.

I would prefer to expose a logger hierarchy that reflects the user-facing module layout rather than the file layout.  For example, a user doesn't care about the difference between `det.pytorch._pytorch_trial` vs `det.pytorch._pytorch_context` but they would care about `det.pytorch` vs `det.storage` (assuming we moved `det.common.storage` to `det.storage` and made it user-facing, that is).  In this case, I think I would like to reorganize all the files related to the generic api into a `det.generic` namespace anyway.

I'll rename it `logger` everywhere.

Reply:
when these are moved into the `generic` subdirectory, they'll all become `determined.generic._something`, and a user can still configure the logging for all of them on `determined.generic` logger in the same fashion. 
The full name provides a couple benefits:
- if the logger name is in the log message, a user will immediately know which file is logging the current error, and can go in and "debug" stuff.
- you can have elaborate logging configs where you set DEBUG level for `determined.generic._checkpointing` but the rest of `determined.generic` is still logging at default INFO/WARNING.

for example, tensorflow log messages include the paths to its `.cc` files, such as: 
```tensorflow/stream_executor/cuda/cuda_dnn.cc:189] Check failed: status == CUDNN_STATUS_SUCCESS (7 vs. 0)Failed to set cuDNN stream. <[]>```